src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,120 @@
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    private static class PersonComparator<T extends Person> implements Comparator<T> {
+
+        @Override
+        public int compare(T firstPerson, T secondPerson) {
+            return Integer.compare(firstPerson.getAge(), secondPerson.getAge());
+        }
+    }
+
+    @Test
+    void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person >() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                return Integer.compare(firstPerson.getAge(), secondPerson.getAge());
+            }
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Иван", "Мельников", 20),
+                new Person("Николай", "Зимов", 30),
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45)
+        )));
+    }
+
+    @Test
+    void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                if(!firstPerson.getLastName().equals(secondPerson.getLastName())) {
+                    return firstPerson.getLastName().compareTo(secondPerson.getLastName());
+                } else {
+                    return firstPerson.getFirstName().compareTo(secondPerson.getFirstName());
+                }
+            }
+        });
+
+        assertThat(persons, is(arrayContaining(
+                new Person("Алексей", "Доренко", 40),
+                new Person("Артем", "Зимов", 45),
+                new Person("Николай", "Зимов", 30),
+                new Person("Иван", "Мельников", 20)
+        )));
+    }
+
+    @Test
+    void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Predicate<Person> personPredicate = person -> ((Integer) 30).equals(person.getAge());
+
+        Person person = FluentIterable.from(persons)
+                                      .firstMatch(personPredicate)
+                                      .get();
+
+        assertThat(person, is(new Person("Николай", "Зимов", 30)));
+    }
+
+    @Test
+    void findFirstWithAge30UsingGuavaAnonymousPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        Predicate<Person> personPredicate = new Predicate<Person>() {
+            @Override
+            public boolean apply(Person person) {
+                return ((Integer) 30).equals(person.getAge());
+            }
+        };
+        Person person = FluentIterable.from(persons)
+                                      .firstMatch(personPredicate)
```

Comment:
And in this method you need to write _GuavaAnonymousPredicate_, so it will be like `Person person = FluentIterable.from(persons).firstMatch(new Predicate<Person>() {...})...`

Reply:
done